### PR TITLE
Compact log proofs

### LINKED ISSUE #19: Enable compact LOG proof for solEVM
Before the enforcer can run an op-code on chain, every VM element needs to be initialized.
Currently the LOG buffer is initialized as follows:

- Full LOG buffer is provided in initialization transaction.
- LOG buffer is hashed and combined with hashes of other VM elements.
- If the VM root hash matches with generated hash, VM has been set up correctly.

This is not efficient, as the LOG buffer is append only, and previous events don't affect the execution. A better approach would be to hash new LOG entries with previous LOG hash, creating a chain of LOGs.

### PR DESCRIPTION
**Pull request by pinkiebell**
Also switch from web3 to ethers.js.
Web3 1.x has decoding errors in emitted events.
Web3 0.x don't support structs and other nice stuff.

Ethers.js is fine and also has better error reporting, types checking.
I had to downgrade ganache because the newer version modifies
transactions midflight, ethers.js catched that and throws an Error.

Fixes #19

### CODE REVIEW

**Comment by pinkiebell** [comment]
@johannbarbie Want to take a look? 😄 
I removed the `EVMLogs.Logs logs` from the struct(s) as it seems we only need the hash of the logs in the verifier steps. Right?

File: test/ethereumRuntime.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { toNum, encodeAccounts, decodeAccounts, decodeLogs, getCode } from './utils';
+import { range, hexRange,toNum, toStr, encodeAccounts, decodeAccounts, decodeLogs, getCode, deployContract } from './utils';
```
Comment (troggy) [inline test/ethereumRuntime.js]:
```suggestion
import { toNum, toStr, encodeAccounts, decodeAccounts, getCode, deployContract } from './utils';
```

File: test/ethereumRuntime.js [hunk 2]
```diff
@@ -55,19 +59,24 @@ contract('Runtime', function () {
...
+          code,
+          callData,
           [pc, 0, blockGasLimit, gasLimit],
-          initialStack, initialMemory, accounts, accountsCode, logs, logsData,
-        );
+          initialStack,
+          initialMemory,
+          accounts,
+          accountsCode,
+          logHash
```
Comment (troggy) [inline test/ethereumRuntime.js]:
```suggestion
          logHash,
```

File: test/fixtures.js [hunk 3]
```diff
@@ -288,75 +288,172 @@ export default [
...
+      gasUsed: 375
+    }
+  },
+  {
+    code: OP.LOG1,
+    stack: [0x567, 4, 2],
+    memory: '0x0102030405060708091011121314151617181920212223242526272829303132',
+    result: {
+      stack: [],
+      logs: [
```
Comment (troggy) [inline test/fixtures.js]: we don't need logs in fixtures anymore, cause we are not using them in tests

**Comment by pinkiebell** [comment]
@johannbarbie @troggy 
Thanks for review & Kudos ^^ .
I pushed another commit to resolve the leftover comments 👍

**Pull request merged**